src/fread.c

```diff
@@ -1604,14 +1609,20 @@ int freadMain(freadMainArgs _args) {
   if (initialBuffRows > INT32_MAX) STOP("Buffer size %lld is too large\n", initialBuffRows);
   nth = imin(nJumps, nth);
 
+
   read:  // we'll return here to reread any columns with out-of-sample type exceptions
+  {
+  if (verbose) {
+    DTPRINT("[11] Read the data\n");
+    DTPRINT("  jumps=[%d..%d), chunk_size=%llu, total_size=%llu\n", jump0, nJumps, (llu)chunkBytes, (llu)(lastRowEnd-pos));
+  }
+  ASSERT(allocnrow <= nrowLimit);
```

Comment:
Need the values of `allocnrow` and `nrowLimit` included in the error message please if assert fails.